sw/device/lib/runtime/core_v_mini_mcu.c

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 EPFL
+// Solderpad Hardware License, Version 2.1, see LICENSE.md for details.
+// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+
+#include "core_v_mini_mcu.h"
+#include <stdint.h>
+
+// functions
+uint32_t * heep_get_data_address_lma(uint32_t* data_address_vma);
+uint32_t * heep_get_flash_address_offset(uint32_t* data_address_lma);
+
+// LINKER symbol, the value is defined in the LINKER script
+extern const uint32_t _lma_vma_data_offset;
+
+// this translates the virtual address to logical address for the FLASH
+uint32_t * heep_get_data_address_lma(uint32_t* data_address_vma){
+
+    uint32_t* data_address_lma = (uint32_t*) ((uint32_t)&(_lma_vma_data_offset) + (uint32_t)(data_address_vma));
```

Comment:
lma is load memory address (logical and virtual are the same).
question (maybe i am right now confused): _lma_vma_data_offset is already a memory offset, isn't it? i.e. you do not need to take the pointer to it (&), do you?

Reply:
true is load memory, I will change it - for what concern the offset, I thought the same but didnt work, let me try again maybe I made a mistake

Reply:
solved it in last commit